test/parallel/test-net-connect-abort-controller.js

```diff
@@ -0,0 +1,110 @@
+const { getEventListeners } = require('events');
+
+const liveConnections = new Set();
+
+server.listen(0, common.mustCall(async () => {
+  const port = server.address().port;
+  const host = 'localhost';
+  const socketOptions = (signal) => ({ port, host, signal });
+  server.on('connection', (connection) => {
+    liveConnections.add(connection);
+    connection.on('close', () => {
+      liveConnections.delete(connection);
+    });
+  });
+
+  const attachHandlers = (socket, res) => {
+    socket.on('error', common.mustCall((err) => {
+      assert.strictEqual(err.name, 'AbortError');
+    }));
+    socket.on('close', () => {
+      res();
+    });
+  };
+
+  function postAbort() {
+    return new Promise((res) => {
+      const ac = new AbortController();
+      const { signal } = ac;
+      const socket = net.connect(socketOptions(signal));
+      attachHandlers(socket, res);
+      assert.strictEqual(getEventListeners(signal, 'abort').length, 1);
+      ac.abort();
+    });
+  }
+
+  function preAbort() {
+    return new Promise((res) => {
+      const ac = new AbortController();
+      const { signal } = ac;
+      ac.abort();
+      const socket = net.connect(socketOptions(signal));
+      attachHandlers(socket, res);
+      assert.strictEqual(getEventListeners(signal, 'abort').length, 0);
+    });
+  }
+
+  function tickAbort() {
+    return new Promise((res) => {
+      const ac = new AbortController();
+      const { signal } = ac;
+      setImmediate(() => ac.abort());
+      const socket = net.connect(socketOptions(signal));
+      attachHandlers(socket, res);
+      assert.strictEqual(getEventListeners(signal, 'abort').length, 1);
+    });
+  }
+
+  function testConstructor() {
+    return new Promise((res) => {
+      const ac = new AbortController();
+      const { signal } = ac;
+      ac.abort();
+      const socket = new net.Socket(socketOptions(signal));
+      assert.strictEqual(getEventListeners(signal, 'abort').length, 0);
+      attachHandlers(socket, res);
+    });
+  }
+
+  function testConstructorPost() {
+    return new Promise((res) => {
+      const ac = new AbortController();
+      const { signal } = ac;
+      const socket = new net.Socket(socketOptions(signal));
+      attachHandlers(socket, res);
+      assert.strictEqual(getEventListeners(signal, 'abort').length, 1);
+      ac.abort();
+    });
+  }
+
+  function testConstructorPostTick() {
+    return new Promise((res) => {
+      const ac = new AbortController();
+      const { signal } = ac;
+      const socket = new net.Socket(socketOptions(signal));
+      attachHandlers(socket, res);
+      assert.strictEqual(getEventListeners(signal, 'abort').length, 1);
+      setImmediate(() => {
+        ac.abort();
+      });
+    });
+  }
+
+  await postAbort();
+  await preAbort();
+  await tickAbort();
+  await testConstructor();
+  await testConstructorPost();
+  await testConstructorPostTick();
+
+  // Killing the net.socket without connecting hangs the server.
```

Comment:
Probably easier to call socket.setTimeout with a low timeout - but probably what you did is also fine

Reply:
#10128